### PR TITLE
Derive `Hash` for `Recipient`.

### PR DESCRIPTION
**Pull request by gibbz00**
Need to store them in a collection that ensures uniqueness :)

### CODE REVIEW

File: age/src/x25519.rs [hunk 1]
```diff
@@ -146,7 +146,7 @@ impl crate::Identity for Identity {
 ///
 /// This recipient type is anonymous, in the sense that an attacker can't tell from the
 /// age-encrypted file alone if it is encrypted to a certain recipient.
-#[derive(Clone, PartialEq, Eq)]
+#[derive(Clone, PartialEq, Eq, Hash)]
```
Comment (str4d) [inline age/src/x25519.rs:149]: Please update `age/CHANGELOG.md` with this addition (change line 14 to add `, Hash` to the existing list I added for your previous PR).

**Review by str4d — CHANGES_REQUESTED** [review]
****

File: age/src/x25519.rs (hunk 1 above)
Comment (gibbz00) [reply age/src/x25519.rs:149]: Fixed!

**Review by str4d — APPROVED** [review]
utACK

**Pull request merged**